support/builder/s3migrate.sh

```diff
@@ -82,15 +82,36 @@ installDeps() {
 
 setBucket() {
     # Configure the bucket name to use for the upload
-    echo ""
-    echo "==========================================================="
-    echo "Please enter a target bucket name and press [ENTER]:"
-    read bucket_name
+    sgroups=("default" "dev" "prod" "acceptance" "live" "blue" "green")
+    for i in "${sgroups[@]}"; do
+        if curl -s localhost:9631/services/builder-api/"${i}" > /dev/null; then
+            bucket_name=$(curl -s localhost:9631/services/builder-api/"${i}" | jq .cfg.s3.bucket_name)
+            if [[ -n $bucket_name ]]; then
+                echo ""
+                echo "We've detected your minio bucket configuration set to: ${bucket_name}!"
+                read -r -p "Would you like to use this minio bucket? [y/N] " response
+                if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+                    echo "Setting bucket to ${bucket_name}"
+                    bucket_name="${bucket_name//\"}"
+                    break
+                fi
+            fi
+        fi
+    done
+
+    if [[ -z ${bucket_name} ]]; then
```

Comment:
Since this script is using `set -u` and `set -e`, it might be a bit safer to give a default: `${bucket_name:-}`.

If the `curl` command on line 87 were to have a nonzero exit status, `bucket_name` may be unset.